FannyAndersson/PayWay
# Feature: Send money frontend && private route component

# Hello!

This PR introduces the component to send money. It is linked with a silly link at the home page, but can also be accessed at [http://localhost:3000/send-money](http://localhost:3000/send-money). 

Project-related things that I also included in this PR:

- A private route component, that behaves like a normal route but redirects to /login if not logged in

- Added --ignore flag to npm start script, so nodemon isn't restarting the backend when changes are made to frontend code.

- Gitignored output of sass-files. To accomplish this, the file must be deleted from the git repo and then ignored. This means that you **must create the folder called css inside frontend/src/ and create the file called style.css inside that folder!**. If you switch back and forth between old branches and new ones, you might have to do it again, so try to keep all your branches as fresh as possible :).

- Updated ContextKeeper to wait on server response before actually rendering anything other than a spinner - this solves a problem where if you render for instance a private route, while a user is waiting for the call to the login API, it is treated as not logged in and redirected before the login has time to take place. This won't be so fun if the server is down, but if the server is down, the our app is quite useless, so it won't be fun anyway.

## What is not included

Any type of success message - it is a separate task. If your transaction is successful, the loading icon will disappear and nothing else will happen - if it is not successful, you will receive some kind of error message.

There are only two error messages - one in case the recipient does not exist, and one more generic error message.

### How to test

Make sure you are logged in and send some money. Verify in the database that your transaction is created. Try to send money to non-existing numbers, try to send too much money. Be as devious as you can and try to find the use cases for which this will break or behave badly.
